internal/multicluster/internal/types/validation_exported_services_ce.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !consulent
+
+package types
+
+import (
+	"fmt"
+	"github.com/hashicorp/consul/internal/resource"
+	multiclusterv1alpha1 "github.com/hashicorp/consul/proto-public/pbmulticluster/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/go-multierror"
+)
+
+func ValidateExportedServicesEnterprise(res *pbresource.Resource, exportedService *multiclusterv1alpha1.ExportedServices) error {
+	var hasSetEnterpriseFeatures bool
+
+	invalidFields := make([]string, 0)
+
+	if res.Id != nil && res.Id.Tenancy != nil && (res.Id.Tenancy.Namespace != "default" || res.Id.Tenancy.Partition != "default") {
+		if res.Id.Tenancy.Namespace != "default" {
+			invalidFields = append(invalidFields, "namespace")
+		} else {
+			invalidFields = append(invalidFields, "partition")
+		}
+		hasSetEnterpriseFeatures = true
+	}
+
+	for _, consumer := range exportedService.Consumers {
+		// in case partition is not default and for the case when consumer has peer
+		// and non blank partition
+		if consumer.GetPartition() != "default" && consumer.GetPartition() != "" {
```

Comment:
We can ensure that partition is properly defaulted in a mutation hook. Mutation hooks run before validations, so we can ensure partition is defaulted to `default` and is never empty. Here is an example https://github.com/hashicorp/consul/blob/main/internal/auth/internal/types/traffic_permissions.go#L30-L52